migration-guide.md

```diff
@@ -0,0 +1,98 @@
+```
+
+### Listen on multiple TCP sockets with non-sequential ports
+
+```shell
+# v1
+./cloud_sql_proxy -instances=<INSTANCE_CONNECTION_NAME>=tcp:6000,<INSTANCE_CONNECTION_NAME2>=tcp:7000
+
+# v2
+./cloudsql-proxy 'INSTANCE_CONNECTION_NAME?port=6000' 'INSTANCE_CONNECTION_NAME2?port=7000'
+```
+
+### Listen on all interfaces
+
+```shell
+# v1
+./cloud_sql_proxy -instances=<INSTANCE_CONNECTION_NAME>=tcp0.0.0.0:6000
+
+# v2
+./cloudsql-proxy --address 0.0.0.0 --port 6000 INSTANCE_CONNECTION_NAME
+```
+
+## Flag Changes
+
+The following table lists in alphabetical order v1 flags and their v2 version.
+
+- 🗓️: Planned
+- ❌: Deprecated
+- 🤔: Unplanned, but has open feature request
+
+| v1                          | v2                    | Notes                                                                                |
+| --------------------------- | --------------------- | ------------------------------------------------------------------------------------ |
+| check_region                | ❌                    |                                                                                      |
+| credential_file             | credentials-file      |                                                                                      |
+| dir                         | unix-socket           |                                                                                      |
+| enable_iam_login            | auto-iam-authn        |                                                                                      |
+| fd_rlimit                   | 🤔                    | [Feature Request](https://github.com/GoogleCloudPlatform/cloudsql-proxy/issues/1258) |
+| fuse                        | 🗓️                    |                                                                                      |
+| fuse_tmp                    | 🗓️                    |                                                                                      |
+| health_check_port           | http-port             |                                                                                      |
+| host                        | sqladmin-api-endpoint |                                                                                      |
+| instances_metadata          | 🤔                    | [Feature Request](https://github.com/GoogleCloudPlatform/cloudsql-proxy/issues/1259) |
+| ip_address_types            | private-ip            | Defaults to public                                                                   |
+| log_debug_stdout            | ❌                    | v2 logs to stdout, errors to stderr by default                                       |
+| max_connections             | max-connections       |                                                                                      |
+| projects                    | ❌                    |                                                                                      |
```

Comment:
The --projects flag probably deserves an explanation about why it is not supported in v2: too much magic.